src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -670,7 +670,15 @@ Transactor::operator()()
     if (ctx_.size() > oversizeMetaDataCap)
         result = tecOVERSIZE;
 
-    if ((result == tecOVERSIZE) || (result == tecKILLED) ||
+    if (isTecClaim(result) && (view().flags() & tapFAIL_HARD))
+    {
+        // If the tapFAIL_HARD flag is set, a tec result
+        // must not do anything
+
+        ctx_.discard();
+        applied = false;
+    }
+    else if ((result == tecOVERSIZE) || (result == tecKILLED) ||
         (isTecClaimHardFail (result, view().flags())))
```

Comment:
strictly from a naming perspective, now that you've added a new flag called `FAIL_HARD`, having a method with the name "HardFail" in it that doesn't actually consider that flag seems misleading - just food for thought.

Reply:
Added the check for tapFAIL_HARD.